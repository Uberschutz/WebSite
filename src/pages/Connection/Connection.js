import React, { Component } from 'react';
import '../../styles/bootstrap.css';
import '../../styles/Connection.css';

import { Link } from 'react-router-dom';
import { displayContent } from '../../utils/translationDisplay';
import axios from 'axios';
import {Alert} from "reactstrap";
import loading from "../../assets/Spinner-1s-70px.gif";

class Connection extends Component {
	constructor(props) {
        super(props);
        this.state = {
        	name: '',
	        surname: '',
	        email: '',
	        password: '',
	        alphaName: false,
	        alphaSurname: false,
	        lang: 'fr',
	        failAuth: false,
            requestSent: false
        };
        this.onChangeEmail = this.onChangeEmail.bind(this);
        this.onChangePass = this.onChangePass.bind(this);
        this._handleKeyPressed = this._handleKeyPressed.bind(this);
        this.connect = this.connect.bind(this);
    }

	componentDidMount() {
		if (this.props.base) {
			const { base: { language } } = this.props;
			console.log(language, this.state.lang, 'kek');
			if (this.state.lang !== language) {
				this.setState({
					lang: language
				})
			}
		}
	}

	componentDidUpdate(prevProps, prevState, snapshot) {
		// console.log(prevProps, this.props);
		this.props.base.language !== prevProps.base.language && this.setState({lang: this.props.base.language}, () => console.log('re'));
	}

	_handleKeyPressed(e) {
		if (e.key === 'Enter') {
			this.connect();
		}
	}

    static isAlpha(char) {
        const re = /^[a-zA-Z]+$/g;
        return re.test(char);
    }

    onChangeEmail(email) {
		this.setState({email: email.target.value});
	}

	onChangePass(pass) {
		this.setState({password: pass.target.value});
	}

	connect() {
		if (this.state.email && this.state.email !== '' && this.state.password !== '') {
            this.setState({requestSent: true});
		    axios.post('/connect', {
		        email: this.state.email,
                passwd: this.state.password
            }).then(response => {
                console.log(response.data);
                this.props.setLogged(true);
                // this.props.setUser(this.state.email, response.data.lastname, response.data.firstname);
                this.props.setAuthToken(response.data.token);
                // this.props.setNewsletter(response.data.newsletter);
                this.props.history.push('/');
            }).catch(err => {
                console.log(err);
                this.setState({failAuth: true}, () => {
                	setTimeout(() => {
                		this.setState({failAuth: false});
	                }, 10 * 1000);
                }).finally(() => {
                    this.setState({requestSent: false});
                });
            });
		}
	}

    render() {
	    let i = 0;
        return (
            <div>
                <br/>
                <div className="card text-white bg-dark box-center">
                    <div className="card-body bg-dark item-align text-align">
                        <h5 className="card-title text-align">
                            {displayContent(this.state.lang, i++, 'connexion')}
                        </h5>
                        <p className="card-text">
                            {displayContent(this.state.lang, i++, 'connexion')}
                        </p>
                        <div className="input-group mb-3">
                            <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroup-sizing-default">Email</span>
                            </div>
                            <input value={this.state.email} onChange={this.onChangeEmail} onKeyPress={this._handleKeyPressed} type="text" className="form-control" aria-label="Email"/>
                        </div>
                        <div className="input-group mb-3">
                            <div className="input-group-prepend">
                                <span className="input-group-text" id="inputGroup-sizing-default">{displayContent(this.state.lang, i++, 'connexion')}</span>
                            </div>
                            <input value={this.state.password} onChange={this.onChangePass} onKeyPress={this._handleKeyPressed} type="password" className="form-control" aria-label="Email"/>
                        </div>
                        <button onClick={this.connect} type="button" className="btn btn-primary">{displayContent(this.state.lang, i++, 'connexion')}</button> <br/>
                        {
                            this.state.requestSent ? <img src={loading} alt="loading"/>
                                : null
                        }
                    </div>
                </div>
	            {
	            	this.state.failAuth ? <Alert color="danger">Ce compte n'existe pas</Alert> : null
	            }
                <br/>
                <span>
                    {displayContent(this.state.lang, i++, 'connexion')}
                </span> <br/>
                <Link to={'/Registration'}>
                    <button type="button" className="btn btn-primary">{displayContent(this.state.lang, i++, 'connexion')}</button>
                </Link>
            </div>
        )
    }
}

export default Connection;