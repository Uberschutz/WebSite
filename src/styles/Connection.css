.box-center {
    align-items: center;
    justify-content: center;
    margin-left: 10%;
    margin-right: 10%;
}

.text-align {
    text-align: center;
}

.item-align {
    align-items: center;
    align-content: center;
    justify-items: center;
    justify-content: center;
}

.form-box {
    width: 50%;
    margin-left: 25%;
}

.responsive-image {
    position: relative;
    max-width: 100%;
}

.form-box-error {
	border: 2px solid red!important;
}

.address {
	font-style: italic;
}