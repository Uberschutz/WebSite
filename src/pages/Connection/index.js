import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'

import * as actions from '../../actions'
import Connection from './Connection'

const mapStateToProps = state => ({
	base: state.base,
})

const mapDispatchToProps = dispatch => ({
	setLanguage: bindActionCreators(actions.setLanguage, dispatch)
})

const ConnectionContainer = connect(mapStateToProps, mapDispatchToProps)(Connection)

export default ConnectionContainer