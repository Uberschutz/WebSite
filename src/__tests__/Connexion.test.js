import React from 'react';
import ReactDOM from 'react-dom';
import Connection from '../pages/Connection/Connection';
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

it('renders Connection without crashing', () => {
    const div = document.createElement('div');
	ReactDOM.render(<Router><Connection lang='fr'/></Router>, div);
    expect(div).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
});
