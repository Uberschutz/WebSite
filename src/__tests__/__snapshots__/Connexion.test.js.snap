// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Connection without crashing 1`] = `
<div>
  <div>
    <div
      class="row responsive-image"
    >
      <div
        class="col"
      >
        <br />
        <div
          class="card text-white bg-dark box-center"
        >
          <div
            class="card-body bg-dark item-align text-align"
          >
            <h5
              class="card-title text-align"
            >
              Authentification
            </h5>
            <p
              class="card-text"
            >
              Merci de rentrer votre email et votre mot de passe pour continuer
            </p>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Email
                </span>
              </div>
              <input
                aria-label="Email"
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Mot de passe
                </span>
              </div>
              <input
                aria-label="Email"
                class="form-control"
                type="password"
                value=""
              />
            </div>
            <a
              href="/"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Continuer
              </button>
            </a>
          </div>
        </div>
        <br />
      </div>
      <div
        class="col"
      >
        <br />
        <div
          class="card text-white bg-dark box-center"
        >
          <div
            class="card-body bg-dark item-align text-align"
          >
            <h5
              class="card-title text-align"
            >
              Inscription
            </h5>
            <p
              class="card-text"
            >
              Merci de remplir ces champs pour continuer
            </p>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Nom
                </span>
              </div>
              <input
                aria-label="Nom"
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Prénom
                </span>
              </div>
              <input
                aria-label="Prénom"
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Email
                </span>
              </div>
              <input
                aria-label="Email"
                class="form-control"
                type="text"
                value=""
              />
            </div>
            <div
              class="input-group mb-3"
            >
              <div
                class="input-group-prepend"
              >
                <span
                  class="input-group-text"
                  id="inputGroup-sizing-default"
                >
                  Mot de passe
                </span>
              </div>
              <input
                aria-label="Email"
                class="form-control"
                type="password"
                value=""
              />
            </div>
            <a
              href="/Profile"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Continuer
              </button>
            </a>
            <br />
          </div>
        </div>
        <br />
      </div>
    </div>
  </div>
</div>
`;
